Pass :index option through to hidden field

mattbrictson opened the pull request:
Currently, if we call `attachment_field(:image, index: 1)`, Refile omits the `:index` option when rendering the hidden field. This means the hidden field and file field are rendered with different names, e.g.:

```
<input type=hidden name=object[image] ...>
<input type=file   name=object[1][image] ...>
```

This is a bug.

The hidden field and file field need to render with identical `name` attributes in order for file upload to work properly. The name is generated by Rails based on certain options, so we need to make sure that both the field and file receive the same options, at least those options that affect the name. `:index` is one of those options. 

This PR fixes this discrepancy by passing the `:index` option to the hidden field so that the output is correct:

```
<input type=hidden name=object[1][image] ...>
<input type=file   name=object[1][image] ...>
```

jnicklas commented:
Nice :) Thank you! It seems like this is a solution to #335? 

I'm not sure if there's any sensible way of testing this, some kind of regression test would be super nice, but maybe not absolutely required for me to merge this.

mattbrictson commented:
@jnicklas I've updated the PR and added a test. There wasn't yet a test for `attachment_helper`, so I added one that covers the `:index` scenario. I hope that is the right place to put it.

jokklan commented:
Ahh couldn't figure this bug out! Awesome really needed that fix, thanks +1

simonwh commented:
:+1: Fantastic, also really needed this.

jnicklas commented:
Awesome :) Thanks!
